src/core/workload.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Base objects for workload operations across VM + K8s charms."""
+from abc import ABC, abstractmethod
+
+from literals import PATHS
+
+
+class ODPaths:
+    """Collection of expected paths for the Opensearch Dashboards workload."""
+
+    def __init__(self):
+        self.conf_path = PATHS["CONF"]
+        self.data_path = PATHS["DATA"]
+        self.binaries_path = PATHS["BIN"]
+        self.logs_path = PATHS["LOGS"]
+
+    @property
+    def data_dir(self) -> str:
+        """The directory where Opensearch Dashboards will store the in-memory database snapshots."""
+        return f"{self.data_path}/data"
+
+    @property
+    def properties(self) -> str:
+        """The main properties filepath.
+
+        Contains all the main configuration for the service.
+        """
+        return f"{self.conf_path}/opensearch_dashboards.yml"
+
+    @property
+    def server_key(self) -> str:
+        """The private-key for the service to identify itself with for TLS auth."""
+        return f"{self.conf_path}/certificates/server.key"
+
+    @property
+    def ca(self) -> str:
+        """The shared cluster CA."""
+        return f"{self.conf_path}/certificates/ca.pem"
+
+    @property
+    def certificate(self) -> str:
+        """The certificate for the service to identify itself with for TLS auth."""
+        return f"{self.conf_path}/certificates/server.pem"
+
+    @property
+    def opensearch_ca(self) -> str:
+        """The certificate for the service to identify itself with for TLS auth."""
+        return f"{self.conf_path}/certificates/opensearch_ca.pem"
+
+
+class WorkloadBase(ABC):
+    """Base interface for common workload operations."""
+
+    paths = ODPaths()
+
+    @abstractmethod
+    def start(self) -> None:
+        """Starts the workload service."""
+        ...
+
+    @abstractmethod
+    def stop(self) -> None:
+        """Stops the workload service."""
+        ...
+
+    @abstractmethod
+    def restart(self) -> None:
+        """Restarts the workload service."""
+        ...
+
+    @abstractmethod
+    def read(self, path: str) -> list[str]:
```

Comment:
```suggestion
    def read(self, path: str) -> List[str]:
```

Reply:
See https://github.com/canonical/opensearch-dashboards-operator/pull/1#discussion_r1557363244, I'd prefer NOT to do this.